src/simulation.cpp

```diff
@@ -686,8 +690,15 @@ void transport_history_based_single_particle(Particle& p)
       p.event_collide();
     }
     p.event_revive_from_secondary();
-    if (!p.alive())
+    if (!p.alive()) {
```

Comment:
This is not the right place to do the scoring. It should probably be in `p.event_collide()`. It should be right near `score_collision_tally` in that function.

Reply:
I had the problem here that we can do the scoring only after the particle and all its progeny have been simulated. That's why I put it in this unusual place. I don't know exactly how we can put this in `p.event_collide()` right now, since when this method is called, it's not yet determined here whether the particle and its progeny have finished simulating. I don't know how we could put this information in here. Maybe I'm on the wrong track here.